activerecord/lib/active_record/environment_check.rb

```diff
@@ -0,0 +1,56 @@
+require 'active_record/scoping/default'
+require 'active_record/scoping/named'
+
+module ActiveRecord
+  # This class is used to create a table that keeps track of which environment
+  # migrations were run in. When a migration is run, its env value such as RAILS_ENV
+  # is inserted in to the `EnvironmentCheck.table_name` we can later check aginst
+  # this value before performing destructive actions.
+  class EnvironmentCheck < ActiveRecord::Base
+    class << self
+      def primary_key
+        nil
+      end
+
+      def table_name
+        "#{table_name_prefix}#{ActiveRecord::Base.environment_check_table_name}#{table_name_suffix}"
+      end
+
+      def index_name
+        "#{table_name_prefix}unique_#{ActiveRecord::Base.environment_check_table_name}#{table_name_suffix}"
+      end
+
+      def table_exists?
+        ActiveSupport::Deprecation.silence { connection.table_exists?(table_name) }
+      end
+
+      # Creates a schema table with columns +environment+ and +version+
+      def create_table
+        if table_exists?
+          return if environment_is_stored?
+
+          connection.change_table(table_name) do |t|
+            t.add_column :environment, :string
```

Comment:
Ahh, I understand now. Good catch, i don't think it can be. I think this is a fragment from when I was updating the SchemaMigration table instead of using a seperate environment check table. I'm updating to remove that logic.